src/lib.rs

```diff
@@ -372,6 +372,19 @@ impl<T> Snarl<T> {
         }
     }
 
+    /// Returns reference to the node and its position.
+    pub fn get_node_pos(&self, idx: NodeId) -> Option<(&T, &Pos2)> {
+        self.nodes.get(idx.0).map(|node| (&node.value, &node.pos))
```

Comment:
Please, write function in the same style

Reply:
Clippy warns about "Manual implementation of `Option::map`" in this function

Reply:
Odd, it doesn't do so in the "mut" version

Reply:
Either way, please, keep style consistent.
If you worry about `clippy::manual_map` - fix it everywhere.